pkg/kube/controllers/webhook_configuration.go

```diff
@@ -144,9 +119,34 @@ func (f *WebhookConfig) setupCertificate() error {
 		f.CaCertificate = caCert.Certificate
 		f.Key = cert.PrivateKey
 		f.Certificate = cert.Certificate
+	} else {
```

Comment:
Do we need an else here? 
You should be able to make sure you have a `data` object in both scenarios.

```
data = data from secret

if secret doesn't exist
  data = generated data
end

set values on f
write to file
```

Reply:
data from secret is map[string]interface, but generated  data is map[string][]uint8. So we use if-else to handle this different.